pkg/operator/vtgate/deployment.go

```diff
@@ -106,6 +107,10 @@ func UpdateDeployment(obj *appsv1.Deployment, spec *Spec) {
 	update.Labels(&obj.Labels, spec.Labels)
 	// Tell Deployment to set the same labels on the Pods it creates.
 	update.Labels(&obj.Spec.Template.Labels, spec.Labels)
+	// Set user labels on the Deployment object.
```

Comment:
Since we only care about setting labels on the pods that the user creates, we don't need to set user labels on the Deployment object, this can be removed.